pennylane/labs/resource_estimation/ops/op_math/symbolic.py

```diff
@@ -178,3 +182,114 @@ def pow_resource_decomp(
     def tracking_name(base_class, z, base_params) -> str:
         base_name = base_class.tracking_name(**base_params)
         return f"Pow({base_name}, {z})"
+
+
+class ResourceExp(Exp, re.ResourceOperator):
+    """Resource class for Exp"""
+
+    @staticmethod
+    def _resource_decomp(base_class, coeff, num_steps, **kwargs):
+
+        while isinstance(base_class, SProd):
+            coeff *= base_class.scalar
+            base_class = base_class.base
+
+        # Custom exponential operator resources:
+        if isinstance(base_class, re.ResourceOperator):
+            try:
+                return base_class.exp_resource_decomp(coeff, num_steps, **kwargs)
+            except re.ResourcesNotDefined:
+                pass
+
+        # PauliRot resource decomp:
+        if (pauli_sentence := base_class.pauli_rep) and math.real(coeff) == 0:
+            if qml.pauli.is_pauli_word(base_class):
+                num_wires = len(base_class.wires)
+                pauli_word = tuple(pauli_sentence.keys())[0]  # only one term in the sum
+                return _resources_from_pauli_word(pauli_word, num_wires)
+
+            scalar = num_steps or 1  # 1st-order Trotter-Suzuki with 'num_steps' trotter steps:
+            return _scale_dict(
+                _resources_from_pauli_sentence(pauli_sentence), scalar=scalar, in_place=True
+            )
+
+        raise re.ResourcesNotDefined
+
+    def resource_params(self):
+        return {
+            "base_class": self.base,
+            "coeff": self.scalar,
+            "num_steps": self.num_steps,
+        }
+
+    @classmethod
+    def resource_rep(cls, base_class, coeff, num_steps, **kwargs):
+        name = f"Exp({base_class.__class__.__name__}, {coeff}, num_steps={num_steps})".replace(
+            "Resource", ""
+        )
+        return re.CompressedResourceOp(
+            cls, {"base_class": base_class, "coeff": coeff, "num_steps": num_steps}, name=name
+        )
+
+    @classmethod
+    def pow_resource_decomp(cls, z, *args, **kwargs) -> Dict[re.CompressedResourceOp, int]:
+        return {cls.resource_rep(base_class, z0 * z, base_params): 1}
+        return super().pow_resource_decomp(z, *args, **kwargs)
+
+
+def _resources_from_pauli_word(pauli_word, num_wires):
+    pauli_string = "".join((str(v) for v in pauli_word.values()))
+    len_str = len(pauli_string)
+
+    if len_str == 0:
+        return {}  # Identity operation has no resources.
```

Comment:
```suggestion
    if len_str == 0:
        return {re.ResourceIdentity.resource_rep(): 1}
```
For more complete tracking.

Reply:
Nice catch! Actually this is a bug! The resources of exp(I*theta*Identity) = global phase!